### PR TITLE
Adding PowerShell module, example, and file-based authentication

### PR DESCRIPTION
**Pull request by nickciaravella**

### CODE REVIEW

File: Samples/PowerShell/ExampleCmdletsUsage.ps1 [hunk 1]
```diff
@@ -0,0 +1,47 @@
+<#
+ 
+.COPYRIGHT
+Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+See LICENSE in the project root for license information.
+
+#>
+
+$ModulePath = "H:\publicdw\Intune-Data-Warehouse\Samples\PowerShell\IntuneDataWarehouseCmdlets.psm1"
```
Comment (adamlandmsft) [inline Samples/PowerShell/ExampleCmdletsUsage.ps1]: Can we either put a placeholder here or use a relative path?

File: Samples/PowerShell/ExampleCmdletsUsage.ps1 [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+else {
+    Write-Host
+    Write-Host "Module Path '$ModulePath' doesn't exist..." -ForegroundColor Red
+    Write-Host
+    break
+}
+
+if (!(Test-IntuneDataWarehouseAuthentication)) {
+
+    Write-Host "To use the Intune Data Warehouse, you must first connect providing a few details..."
```
Comment (adamlandmsft) [inline Samples/PowerShell/ExampleCmdletsUsage.ps1]: "you must first provide a few details"

File: Samples/PowerShell/ExampleCmdletsUsage.ps1 [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    $User = Read-Host "Enter the UPN of the user to authenticate as"
+    $ApplicationId = Read-Host "Enter the application ID of an AzureAD app that has access to the API"
```
Comment (adamlandmsft) [inline Samples/PowerShell/ExampleCmdletsUsage.ps1]: Let's use the specific name of the app/role

File: Samples/PowerShell/ExampleCmdletsUsage.ps1 [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    $WarehouseURL = Read-Host "Enter the Intune Data Warehouse provided in the Azure portal"
```
Comment (adamlandmsft) [inline Samples/PowerShell/ExampleCmdletsUsage.ps1]: Intune Data Warehouse URL

File: Samples/PowerShell/README.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+The module exposes 3 basic cmdlets which should cover all of the interaction with the API that are needed.
+
+### Connect-IntuneDataWarehouse
+This cmdlet establishes a connection to the Intune Data Warehouse by authenticating and caching information that will be needed in subsequent requests. These include:
+
+| Parameter     | Description           | Example  |
+| ------------- |-------------| -----|
+| User               | The UPN of the user for authentication | nick@example.com |
+| ApplicationId      | The application Id of an application in AzureAD with authorization to the API | 4184c61a-e324-4f51-83d7-022b6a82b991 |
```
Comment (adamlandmsft) [inline Samples/PowerShell/README.md]: Instead of API we should use the specific app/role naming

File: Samples/PowerShell/README.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+| DataWarehouseURL   | The URL to the data warehouse for your tenant. This can be found in the Azure portal. | https://fef.msua04.manage.microsoft.com/ReportingService/DataWarehouseFEService?api-version=beta |
```
Comment (adamlandmsft) [inline Samples/PowerShell/README.md]: "The Intune Data Warehouse URL for your tenant."

File: Samples/PowerShell/ExampleCmdletsUsage.ps1 (hunk 1 above)
Comment (nickciaravella) [inline Samples/PowerShell/ExampleCmdletsUsage.ps1]: Good idea. Updated it to be the current script execution directory by default.

File: Samples/PowerShell/ExampleCmdletsUsage.ps1 (hunk 3 above)
Comment (nickciaravella) [inline Samples/PowerShell/ExampleCmdletsUsage.ps1]: Using: "... that has access to get data warehouse information from the Microsoft Intune API"

The AzureAD app exposes the role on Microsoft Intune API as: "Get data warehouse information from Microsoft Intune"

File: Samples/PowerShell/ExampleCmdletsUsage.ps1 (hunk 4 above)
Comment (nickciaravella) [inline Samples/PowerShell/ExampleCmdletsUsage.ps1]: nice catch

File: Samples/PowerShell/README.md (hunk 6 above)
Comment (nickciaravella) [inline Samples/PowerShell/README.md]: udpated.

File: Samples/PowerShell/README.md (hunk 5 above)
Comment (nickciaravella) [inline Samples/PowerShell/README.md]: updated.

**Pull request merged**